versioning/BRM/extentmap.cpp

```diff
@@ -2290,68 +2762,76 @@ LBID_t ExtentMap::_createColumnExtent_DBroot(uint32_t size, int OID,
+                    if (fExtentMap[i].status == EXTENTOUTOFSERVICE)
+                        bSegsOutOfService = true;
+
+                    TargetDbRootSegsMapIter iter =
+                        targetDbRootSegs.find(fExtentMap[i].segmentNum);
+
+                    if (iter == targetDbRootSegs.end())
+                    {
+                        targetDbRootSegs.insert(
+                            TargetDbRootSegsMap::value_type(
+                                fExtentMap[i].segmentNum,
+                                fExtentMap[i].blockOffset) );
+                    }
+                    else
                     {
-                        if (fExtentMap[i].segmentNum > partHighSeg)
+                        if (fExtentMap[i].blockOffset > iter->second)
                         {
-                            partHighSeg = fExtentMap[i].segmentNum;
+                            iter->second = fExtentMap[i].blockOffset;
                         }
+                    }
+                } // loop over em idents
+            } // current dbroot == target dbroot
+            else
+            {
+                // 4. Track hi seg for hwm+1 partition
+                auto emIdentsNext = fPExtMapIndexImpl_->find(dbRootFromList, OID, targetDbRootPartNext);
+                for (auto i : emIdentsNext)
+                {
+                    if (fExtentMap[i].segmentNum > partHighSegNext)
+                    {
+                        partHighSegNext = fExtentMap[i].segmentNum;
+                    }
+                }
 
-                        // 6. Save list of seg files in target DBRoot/Partition,
-                        //    along with the highest fbo for each seg file
-                        if (fExtentMap[i].dbRoot == dbRoot)
-                        {
-                            if (fExtentMap[i].status == EXTENTOUTOFSERVICE)
-                                bSegsOutOfService = true;
-
-                            TargetDbRootSegsMapIter iter =
-                                targetDbRootSegs.find(fExtentMap[i].segmentNum);
-
-                            if (iter == targetDbRootSegs.end())
-                            {
-                                targetDbRootSegs.insert(
-                                    TargetDbRootSegsMap::value_type(
-                                        fExtentMap[i].segmentNum,
-                                        fExtentMap[i].blockOffset) );
-                            }
-                            else
-                            {
-                                if (fExtentMap[i].blockOffset > iter->second)
-                                {
-                                    iter->second = fExtentMap[i].blockOffset;
-                                }
-                            }
-                        }
+                // 5. Track hi seg for hwm partition
+                auto emIdents = fPExtMapIndexImpl_->find(dbRootFromList, OID, targetDbRootPart);
```

Comment:
We got `emIdents` here and iterating over `emIdentsNext` on the next line, is it correct?

Reply:
Not at all. Looks like a bug. Thx a lot!

Reply:
Fixed it in a separate PR.